src/components/huskelapp/HuskelappModal.tsx

```diff
@@ -33,88 +52,146 @@ const ModalContent = styled(Modal.Body)`
+      case Oppfolgingsgrunn.FOLG_OPP_FRA_LEGE:
+        return texts.oppfolgingsgrunn.folgOppFraLege;
+      case Oppfolgingsgrunn.FOLG_OPP_FRA_ARBEIDSGIVER:
+        return texts.oppfolgingsgrunn.folgOppFraArbeidsgiver;
+      case Oppfolgingsgrunn.TA_KONTAKT:
+        return texts.oppfolgingsgrunn.taKontakt;
+      case Oppfolgingsgrunn.VURDER_TILTAK_BEHOV:
+        return texts.oppfolgingsgrunn.vurderTiltakBehov;
+      case Oppfolgingsgrunn.ANNEN_OPPFOLGNING:
+        return texts.oppfolgingsgrunn.annenOppfolgning;
+    }
   };
 
-  const handleOppdaterHuskelapp = () => {
-    const huskelappDto: HuskelappRequestDTO = { tekst: tekst };
-    oppdaterHuskelapp.mutate(huskelappDto, {
-      onSuccess: () => toggleOpen(false),
-    });
+  const handleOnSubmit = (e: any) => {
+    e.preventDefault();
+    if (oppfolgingsgrunn !== undefined) {
+      const huskelappDto: HuskelappRequestDTO = {
+        oppfolgingsgrunn: oppfolgingsgrunn,
+      };
+      oppdaterHuskelapp.mutate(huskelappDto, {
+        onSuccess: () => toggleOpen(false),
+      });
+    } else {
+      setIsFormError(true);
+      return undefined;
+    }
+  };
+
+  const handleOppfolgingsgrunnChange = (oppfolgingsgrunn: Oppfolgingsgrunn) => {
+    setOppfolgingsgrunn(oppfolgingsgrunn);
+    setIsFormError(false);
   };
 
   const handleRemoveHuskelapp = (uuid: string) => {
     removeHuskelapp.mutate(uuid, {
       onSuccess: () => toggleOpen(false),
     });
   };
+  const isSavedHuskelapp = !!huskelapp;
 
   return (
-    <Modal
-      closeOnBackdropClick
-      className="px-6 py-4 w-full max-w-[50rem]"
-      aria-label={"huskelapp"}
-      open={isOpen}
-      onClose={() => toggleOpen(false)}
-      header={{ heading: texts.header }}
-    >
-      <ModalContent>
-        {isLoading && <HuskelappSkeleton />}
-        {isSuccess && (
-          <Textarea
-            label={texts.textAreaLabel}
-            hideLabel
-            defaultValue={huskelapp?.tekst}
-            onChange={oppdaterTekst}
-          />
-        )}
-      </ModalContent>
-      <Modal.Footer>
-        <Button
-          variant="primary"
-          onClick={handleOppdaterHuskelapp}
-          loading={oppdaterHuskelapp.isLoading}
-          disabled={isLoading}
-        >
-          {texts.save}
-        </Button>
-        <Button variant="secondary" onClick={() => toggleOpen(false)}>
-          {texts.close}
-        </Button>
-        {huskelapp && (
-          <Tooltip content={texts.removeTooltip}>
+    <form onSubmit={handleOnSubmit}>
+      <Modal
+        closeOnBackdropClick
+        className="px-6 py-4 w-full max-w-[50rem]"
+        aria-label={"huskelapp"}
+        open={isOpen}
+        onClose={() => toggleOpen(false)}
+        header={{ heading: texts.header }}
+      >
+        <ModalContent>
+          {isLoading && <HuskelappSkeleton />}
+          {isSuccess &&
+            (isSavedHuskelapp ? (
+              <BodyShort>{oppfolgingsgrunnToText()}</BodyShort>
+            ) : (
```

Comment:
Jeg må få på plass visning av `tekst` her for å kunne vise huskelapper som henger igjen fra gammelt format.